app/models/mixins/virtual_total_mixin.rb

```diff
@@ -16,7 +16,7 @@ module ClassMethods
     #
     def virtual_total(name, relation)
       define_method(name) do
-        send(relation).count
+        send(relation).try(:size)
```

Comment:
@kbrock Right, the size calls either count or length, depending if you preloaded the objects. Count will always make the SQL query.

So doing e.g.
ems.floating_ips # does sql query
ems.floating_ips.size # just counts length of the array without sql query
ems.floating_ips.count # does sql query every time we call it

So size should be a bit smarter. :-) But it depends if we can use it here.